docs/src/app/components/pages/customization/using-context.md

```diff
@@ -0,0 +1,137 @@
+## Using Context
+
+The `MuiThemeProvider` component simply adds the `muiTheme` object and `uniqueIdGen` function to context.
+If you prefer using context directly instead, you can follow this pattern:
+
+### Pass Theme
+
+Pass theme down the context:
+
+```js
+import React from 'react';
+import baseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import AppBar from 'material-ui/AppBar';
+
+class Main extends React.Component {
+  getChildContext() {
+    return {muiTheme: getMuiTheme(baseTheme)};
+  }
+
+  render () {
+    return <AppBar title="My AppBar" />;
+  }
+}
+
+Main.childContextTypes = {
+  muiTheme: PropTypes.object.isRequired,
+};
+
+export default Main;
+```
+
+Get theme whenever you need to use it in your own components:
+
+```js
+import React from 'react';
+
+class DeepDownTheTree extends React.Component {
+  render () {
+    return (
+      <span style={{color: this.context.muiTheme.palette.textColor}}>
+        Hello World!
+      </span>
+    );
+  }
+}
+
+DeepDownTheTree.contextTypes = {
+  muiTheme: PropTypes.object.isRequired,
+};
+
+export default DeepDownTheTree;
+```
+
+### Pass UniqueId Generator
+
+It should be noted that if you are using components based on `TextField`, `uniqueIdGen` **must** be provided in its context.
+This property is a function of type `() => String` to generate uniqueId for input to make the floating label focusing the input.
+
+```js
+import React, {PropTypes} from 'react';
+import getMuiTheme from 'material-ui/styles/getMuiTheme';
+import getUniqueIdGenerator from 'material-ui/utils/getUniqueIdGenerator';
+import TextField from 'material-ui/TextField';
+import AutoComplete from 'material-ui/AutoComplete';
+
+class Main extends React.Component {
+  state = {uniqueIdGen: this.props.uniqueIdGen || getUniqueIdGenerator()}
+  
+  getChildContext() {
+    return {
+      muiTheme: getMuiTheme(),
+      uniqueIdGen: this.state.uniqueIdGen,
+    };
+  }
+
+  render () {
+    return (
+      <div>
+        <TextField />
+        <AutoComplete />
+      </div>
+    );
+  }
+}
+
+Main.propTypes = {
+  uniqueIdGen: PropTypes.func.isRequired,
+};
+
+Main.childContextTypes = {
+  muiTheme: PropTypes.object.isRequired,
+  uniqueIdGen: PropTypes.func.isRequired,
+};
+
+export default Main;
+```
+
+If you want to use multiple contexts, you should ensure their `uniqueIdGen` do not generate same ids.
+
```

Comment:
@mbrookes It's useful when there are multiple context provider components in component tree. For example, if you want to compare different themes.

```javascript
import React from 'react';
import darkBaseTheme from 'material-ui/styles/baseThemes/darkBaseTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import getUniqueIdGenerator from 'material-ui/utils/getUniqueIdGenerator';

const CompareTheme = () => {
  return (
    <div>
      <MuiThemeProvider
        muiTheme={getMuiTheme(lightBaseTheme)}
        uniqueIdGen={getUniqueIdGenerator('main-one')} >
        <App />
      </MuiThemeProvider>
      <MuiThemeProvider
        muiTheme={getMuiTheme(darBaseTheme)}
        uniqueIdGen={getUniqueIdGenerator('main-two')} >
        <App />
      </MuiThemeProvider>
    </div>
  );
}

export default CompareTheme;
```